.github/workflows/consistency.yml

```diff
@@ -0,0 +1,39 @@
+---
+name: Check .cabal Consistency
+
+on: [push, pull_request]
+
+jobs:
+  check-consistency:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Save original cabal file
+        run: cp ./*.cabal original.txt # Assuming there is only one cabal file in the project
+
+      - name: Set up Haskell
+        uses: haskell-actions/setup@v2
+        with:
+          ghc-version: "latest"
+          enable-stack: true
+          stack-no-global: true
+
+      - name: Install hpack
+        run: |
+          stack install hpack
+          echo "/root/.local/bin" >> "$GITHUB_PATH"
+
+      - name: Generate .cabal file from package.yaml
+        run: hpack
+
+      - name: Remove comments and blank lines for comparison
+        run: |
+          for file in ./*.cabal; do
+            grep -v "^--" "$file" | grep -v "^$" > "generated.txt"
```

Comment:
Is there redundancy between this line and the `grep`ing that happens again below on `generated.txt`?

Reply:
Yes, I think so. I refactored the workflow a bit now.